api/config/config.go

```diff
@@ -76,30 +78,24 @@ type UIConfig struct {
 	StaticPath string `validated:"required"`
 	IndexPath  string `validated:"required"`
 
-	FeastCoreAPI        string `json:"REACT_APP_FEAST_CORE_API"`
-	MerlinAPI           string `json:"REACT_APP_MERLIN_API"`
-	TuringAPI           string `json:"REACT_APP_TURING_API"`
 	ClockworkUIHomepage string `json:"REACT_APP_CLOCKWORK_UI_HOMEPAGE"`
-	FeastUIHomepage     string `json:"REACT_APP_FEAST_UI_HOMEPAGE"`
 	KubeflowUIHomepage  string `json:"REACT_APP_KUBEFLOW_UI_HOMEPAGE"`
-	MerlinUIHomepage    string `json:"REACT_APP_MERLIN_UI_HOMEPAGE"`
-	TuringUIHomepage    string `json:"REACT_APP_TURING_UI_HOMEPAGE"`
 }
 
 // Transform env variables to the format consumed by koanf.
 // The variable key is split by the double underscore ('__') sequence,
 // which separates nested config variables, and then each config key is
-// converted to camel-case.
+// converted to lower camel-case.
 //
 // Example:
 //
 //	MY_VARIABLE => MyVariable
-//	VARIABLES__ANOTHER_VARIABLE => Variables.AnotherVariable
+//	VARIABLES__ANOTHER_VARIABLE => variables.anotherVariable
 func envVarKeyTransformer(s string) string {
 	parts := strings.Split(strings.ToLower(s), "__")
 	transformed := make([]string, len(parts))
 	for idx, key := range parts {
-		transformed[idx] = strcase.ToCamel(key)
+		transformed[idx] = strcase.ToLowerCamel(key)
```

Comment:
This is because of https://github.com/caraml-dev/helm-charts/pull/117/files. I decided to use lowerCamelCase in the ConfigMap, to make it consistent with the lowerCamelCase we use in the Helm value file.

Now, in order to make env var values override configuration read from YAML files, the case of the config keys must be the same. I.e. if your config file contains the key `database.user`, then your env var `DATABASE__PASSWORD` should be stored into the key `database.password`, and not `Database.Password`. Otherwise, the value from `Database.Password` key won't override the value stored at `database.password` key and the result wouldn't be consistent. Hope this explanation makes sense :)